### DISCUSSION TITLE
Is there an API or a configuration to rewrite the untitled empty hints text?

### DISCUSSION THREAD

**Discussion by WJ-Yuan** [microsoft/vscode-discussions#1154] [Extension Development QnA] [answered]
As the title, I want to customize the untitled empty hints text. But I just get "workbench.editor.empty.hint" setting item to display or hidden it.
![image](https://github.com/microsoft/vscode-discussions/assets/59187817/f31f99c2-07a6-4902-a7a5-45832f49739a)

**Comment by alefragnani** [comment, marked as answer]
Hi @WJ-Yuan ,

It seems there is no API available yet. Looking at VS Code source, you will notice the value is hardcoded

https://github.com/microsoft/vscode/blob/cc3429872df5e655abd12c233a9defbd15b0987d/src/vs/workbench/contrib/codeEditor/browser/emptyTextEditorHint/emptyTextEditorHint.ts#L369-L375

And there is also this:

https://github.com/microsoft/vscode/blob/cc3429872df5e655abd12c233a9defbd15b0987d/src/vs/workbench/contrib/codeEditor/browser/emptyTextEditorHint/emptyTextEditorHint.ts#L221

But the `IInlineChatSessionProvider` doesn't seem to have a public / proposed API yet so, you can't provide a hint on your extension. On the other hand, GitHub Copilot already provides a custom hint text, which means there is a good change this API to be made public/proposed in a short period of time

Hope this helps

**Reply by WJ-Yuan** [reply]
Hello, @alefragnani 

Thank you very much for your reply. 

It's a pity that it can't be changed. 

And it would be great if later versions could provide an API to modify it like `MarkdownString`.
